charts/vector/templates/_pod.tpl

```diff
@@ -149,8 +149,16 @@ volumes:
     emptyDir: {}
 {{- end }}
   - name: config
-    configMap:
-      name: {{ if .Values.existingConfigMap }}{{ .Values.existingConfigMap }}{{ else }}{{ template "vector.fullname" . }}{{ end }}
+    projected:
+      sources:
+        - configMap:
+            name: {{ if .Values.existingConfig }}{{ .Values.existingConfig }}{{ else }}{{ template "vector.fullname" . }}{{ end }}
+{{- with .Values.extraConfigs }}
+  {{- range . }}
+        - configMap:
+            name: {{ . }}
+  {{- end }}
+{{- end }}
```

Comment:
The benefit of using `projected` volumes is that you can group a number of sources (ConfigMaps, Secrets, etc) into a single volume, which makes things neater in the templating.

With a `projected` volume rather than ranging and creating `n` volumes for each configmap, we have one volume with `n` configmaps (or secrets if needed).

It's a cleaner abstraction for what we're trying accomplish.